contracts/YieldNotionalLever.sol

```diff
@@ -0,0 +1,510 @@
+    ///         2. Flashloan `borrowAmount` of USDC/DAI from the join
+    ///         3. Deposit USDC/DAI into notional market to get relevant fCash
+    ///         4. Pour the received fCash to borrow fyToken
+    ///         5. Buy base using the borrowed fyToken to pay back the flash loan
+    ///         Here are the steps for ETH:
+    ///         1. Flashloan `borrowAmount` of WETH from the join
+    ///         2. Withdraw WETH to get ETH
+    ///         3. Deposit ETH into notional market to get relevant fCash
+    ///         4. Pour the received fCash to borrow fyToken
+    ///         5. Buy base using the borrowed fyToken to pay back the flash loan
+    /// @param baseAmount The amount of own liquidity to supply.
+    /// @param borrowAmount The amount of additional liquidity to borrow.
+    /// @param seriesId The series to create the vault for.
+    function invest(
+        bytes6 seriesId,
+        bytes6 ilkId,
+        uint256 baseAmount,
+        uint256 borrowAmount
+    ) external payable returns (bytes12 vaultId) {
+        (vaultId, ) = ladle.build(seriesId, ilkId, 0);
+
+        if (ilkInfo[ilkId].currencyId == 0) {
+            INotionalJoin notionalJoin = INotionalJoin(
+                address(ladle.joins(ilkId))
+            );
+            notional.setApprovalForAll(address(notionalJoin), true);
+            IlkInfo storage ilkInfoTemp = ilkInfo[ilkId];
+            ilkInfoTemp.join = FlashJoin(notionalJoin.underlyingJoin());
+            ilkInfoTemp.maturity = notionalJoin.maturity();
+            ilkInfoTemp.currencyId = notionalJoin.currencyId();
+        }
+        // Since we know the sizes exactly, packing values in this way is more
+        // efficient than using `abi.encode`.
+        //
+        // Encode data of
+        // OperationType    1 byte      [0:1]
+        // seriesId         6 bytes     [1:7]
+        // vaultId          12 bytes    [7:19]
+        // ilkId            6 bytes     [19:25]
+        // baseAmount       32 bytes    [25:57]
+        bytes memory data = bytes.concat(
+            bytes1(uint8(uint256(Operation.BORROW))),
+            seriesId,
+            ilkId,
+            vaultId,
+            bytes32(baseAmount)
+        );
+        bool success;
+        IlkInfo memory info = ilkInfo[ilkId];
+        IERC20 token = IERC20(info.join.asset());
+
+        if (ilkInfo[ilkId].currencyId != 1) {
+            // Since notional accepts only ETH we don't have to do any transfer from the user
+            // Transfer the underlying USDC/DAI already approved by the user
+            token.safeTransferFrom(msg.sender, address(this), baseAmount);
+        } else {
+            if (msg.value == 0) {
+                // Transfer weth from user
+                token.safeTransferFrom(msg.sender, address(this), baseAmount);
+                // Convert weth to eth
+                weth.withdraw(baseAmount);
+            }
+        }
```

Comment:
Is this conditional right? Wouldn't it be:
```
        if (ilkInfo[ilkId].currencyId != 1) {
            if (msg.value == 0) {
                // Transfer weth from user
                token.safeTransferFrom(msg.sender, address(this), baseAmount);
                // Convert weth to eth
                weth.withdraw(baseAmount);
            } else {
                // Since notional accepts only ETH we don't have to do any transfer from the user
                // Transfer the underlying USDC/DAI already approved by the user
                token.safeTransferFrom(msg.sender, address(this), baseAmount);
            }
        } 
```